### PR TITLE
[ Enhancement ] Card Template Editor UI

### LINKED ISSUE #7951: Card Template Editor UI text entry areas unusably small
**Is your feature request related to a problem? Please describe.**
The windows of the three editing pages (front, style, and back) are too small. When there is a lot of content, it takes a long time to slide down to find the corresponding code. It is very troublesome. It is very inconvenient to adjust the page!
**Describe the solution you'd like**
I hope to be able to add pop-up buttons separately, click to open a separate editing window, and edit the page code in full screen mode to facilitate viewing the corresponding code
**Describe alternatives you've considered**
Add a pop-up button next to the small box, click to open a separate editing window, and edit the page code in full-screen mode

**Additional context**

### PR DESCRIPTION
**Pull request by Akshay0701**
## Fixes
Fixes #7951

## Purpose
The problem is the compact view of the current Card Template Editor,
So the solution I think would be changing the current UI to matching with AnkiDesktop Card Template Editor.

### So UI may look like this: -
Capture

### CODE REVIEW

**Comment by Akshay0701** [comment]
> com.ichi2.anki.tests.ContentProviderTest > testInsertAndRemoveNote[0][TestingAVD(AVD) - 11] FAILED 
> Test failed to run to completion. Reason: 'Instrumentation run failed due to 'Process crashed.''. Check device logcat for details
> Tests on TestingAVD(AVD) - 11 failed: Instrumentation run failed due to 'Process crashed.'

what should I do to fix this ?

File: AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.java [hunk 1]
```diff
@@ -369,21 +377,48 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+            mEditorEditText = mainView.findViewById(R.id.editor_editText);
+            mCurrentEdittextId = R.id.front_edit;
+            mEditorEditText.setText(mFrontString);
+            mCurrentEdtiorTitle.setText(R.string.card_template_editor_front);
+
+
+            // setting radiobutton
+            RadioGroup radioGroup1 = (RadioGroup) mainView.findViewById(R.id.radioGroup);
+            radioGroup1.setOnCheckedChangeListener((radioGroup, i) -> {
+                if (i == R.id.styling_edit) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.java:396]: should work here outside the if: `mCurrentEdittextId = i;`

You'll need a better name for `i`

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Looks good!

* Can you try having the centre text aligned it next to the button, not centred
* it feels we need a little more margin/padding between the buttons
* Could we try the text: "Front/Styling/Back" here, to provide the text more room to breathe. I don't know how this would look
* Have you tried the accessibility scanner

* What does it look like if the radiobutton labels contain too much text? New line or truncated? Is this acceptable?
* Is the content centre-aligned? I think top-aligned would be best
* What happens when the content contains more than a page? centre or top aligned?

**Comment by david-allison-1** [comment]
Additional thoughts: 

* Can we turn off autocorrect for the text field
* Can we use a "code" font? Droid Sans Mono

**Comment by mikehardy** [comment]
Oh this is exciting! The previous UI was just unusable, a real disaster (and I was partly responsible!!)

This is a big step forward

I'll review it too in my next batch

**Comment by Akshay0701** [comment]
> Could we try the text: "Front/Styling/Back" here, to provide the text more room to breathe. I don't know how this would look

- I think it requires to use "Front/Styling/Back" or otherwise layout may look like this -:


Capture

> Have you tried the accessibility scanner

- Oh Thanks!, I used it right now and its pretty useful, this app suggested me UI changes and i did that ,now UI look like this
- I changed content align to top ,so i think its much better now.



> Can we turn off autocorrect for the text field

Yes, we can do this, my idea is instead of saving all content after each textChanged ,we can save content of textView when user change the view ,like if user did some changes in front edit text and when user click to back radioButton at that time we will save our front changes ,what do you say ?

**Comment by david-allison-1** [comment]
This looks so much better! 

> > Can we turn off autocorrect for the text field
> 
> Yes, we can do this, my idea is instead of saving all content after each textChanged ,we can save content of textView when user change the view ,like if user did some changes in front edit text and when user click to back radioButton at that time we will save our front changes ,what do you say ?

I don't think this is relevant to autocorrect, it should be an attribute on the EditText

Putting the value in the model is fine, as long as it's not persisted to the database

I think the radio button area can be made a little less tall, but go with the accessibility scanner

I do think we should consider a different font choice here (droid sans mono), but other than that it looks great!

**Comment by Akshay0701** [comment]
Thanks !, And if there is any other UI-related work is pending, then let me know, I would love to do that.

**Comment by Akshay0701** [comment]
What a design now,Thanks to @infinyte7 ❤️

**Comment by infinyte7** [comment]
Its looks good to me. Also view it in night mode, how it works.

**Comment by mikehardy** [comment]
This is so much more usable, very cool. The only thing that worries me now is that people tend to get lost in UIs with two sets of tabs. I think AnkiMobile avoided that by having each card take you into the equivalent of it's own activity. That would be a really big change I think and not something I want to contemplate here even if maybe it is the "correct" thing to do for UX

What I do think might help is just altering the appearance of the top tab mar so that the whole tab bar had a line across it so it was clear that the entire bottom was a content pane, and the selected one then just stood out by being thicker or similar ?

**Comment by mikehardy** [comment]
Or maybe that's worse :smile: - I'm bad at design

**Comment by Akshay0701** [comment]
ln70bocwvxfztwt9321p

I think Mike,we should keep our UI appearance like this, most of the Implementation done like this,it will separate the main content from tab layout and it will make little thicker

**Review by david-allison-1 — DISMISSED** [review]
Looks great! Thanks

**Comment by Akshay0701** [comment]
but there is an error in Emulator Tests / Android Emulator Test (pull_request) ?

**Comment by Akshay0701** [comment]
Test passing in my local machine, what should I do?
Capture

**Comment by mikehardy** [comment]
Hmm unsure about the test then - I just kicked off Emulator Tests again...

**Review by david-allison-1 — APPROVED** [review]
LGTM then

**Review by mikehardy — APPROVED** [review]
Just in time for release-2.15 - this is a huge UI improvement for this release, I think it may have been the most painful part of the app to use, even if most people used it rarely, it was awful when they had to. Thanks!

**Pull request merged**